### ISSUE TITLE
error[invalid-type-arguments] for TypeAlias involving ParamSpec Concatentate

### ISSUE THREAD

**Issue by GalHorowitz** [astral-sh/ty#2024]
### Summary

Example code:
```python
from typing import Callable, Concatenate, ParamSpec, TypeAlias, TypeVar

T = TypeVar("T")
P = ParamSpec("P")

A: TypeAlias = Callable[Concatenate[bool, P], T]
B: TypeAlias = T | A[P, T]
C: TypeAlias = B[T, []]
```

Result of running `ty check`:
```
error[invalid-type-arguments]: No type argument provided for required type variable `T`
 --> ty_test.py:8:16
  |
6 | A: TypeAlias = Callable[Concatenate[bool, P], T]
7 | B: TypeAlias = T | A[P, T]
8 | C: TypeAlias = B[T, []]
  |                ^^^^^^^^
  |
info: rule `invalid-type-arguments` is enabled by default
```

### Version

ty 0.0.2 (42835578d 2025-12-16)

**Comment by AlexWaygood** [comment]
Thanks for the report! We don't support `Concatenate` yet; I've added this as a child issue of #1535.
